File: app/src/main/java/it/cammino/risuscito/viewmodels/ConsegnatiViewModel.java
package it.cammino.risuscito.viewmodels;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;

import java.util.ArrayList;
import java.util.List;

import it.cammino.risuscito.database.RisuscitoDatabase;
import it.cammino.risuscito.database.entities.Canto;
import it.cammino.risuscito.items.CheckableItem;
import it.cammino.risuscito.items.SimpleItem;

public class ConsegnatiViewModel extends AndroidViewModel {

  public boolean editMode;

  public List<CheckableItem> titoliChoose = new ArrayList<>();

  public List<SimpleItem> titoli = new ArrayList<>();

  private LiveData<List<Canto>> mIndexResult;

  private RisuscitoDatabase mDb;

  public ConsegnatiViewModel(Application application) {
    super(application);
  }

  public LiveData<List<Canto>> getIndexResult() {
    if (mIndexResult == null) mIndexResult = new MutableLiveData<>();
    return mIndexResult;
  }

  public void createDb() {
    mDb = RisuscitoDatabase.getInstance(getApplication());
    // Receive changes
    subscribeToDbChanges();
  }

  private void subscribeToDbChanges() {
    mIndexResult = mDb.consegnatiDao().getLiveConsegnati();
  }
}

File: app/src/main/java/it/cammino/risuscito/viewmodels/ConsegnatiViewModel.kt
package it.cammino.risuscito.viewmodels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.map
import it.cammino.risuscito.R
import it.cammino.risuscito.database.RisuscitoDatabase
import it.cammino.risuscito.items.CheckableItem
import it.cammino.risuscito.items.CheckablePassageItem
import it.cammino.risuscito.items.NotableItem
import it.cammino.risuscito.items.notableItem
import it.cammino.risuscito.utils.Utility
import it.cammino.risuscito.utils.extension.useOldIndex

class ConsegnatiViewModel(application: Application) : AndroidViewModel(application) {

    var editMode: MutableLiveData<Boolean> = MutableLiveData(false)

    var titoliChoose: List<CheckableItem> = ArrayList()
    var titoliChooseFiltered: List<CheckableItem> = ArrayList()

    var titoli: List<NotableItem> = ArrayList()
    var passaggi = ArrayList<CheckablePassageItem>()

    var mIndexResult: LiveData<List<NotableItem>>? = null

    var mIdConsegnatoSelected: Int = 0
    var mIdCantoSelected: Int = 0

    init {
        val useOldIndex = application.useOldIndex()
        val mDb = RisuscitoDatabase.getInstance(getApplication())
        mIndexResult = mDb.consegnatiDao().liveConsegnati().map { canti ->
            val newList = ArrayList<NotableItem>()
            canti.forEach {
                newList.add(
                    notableItem {
                        setTitle = Utility.getResId(it.titolo, R.string::class.java)
                        setPage = Utility.getResId(
                            if (useOldIndex) it.pagina + Utility.OLD_PAGE_SUFFIX else it.pagina,
                            R.string::class.java
                        )
                        setSource = Utility.getResId(it.source, R.string::class.java)
                        setColor = it.color
                        id = it.id
                        idConsegnato = it.consegnato
                        numPassaggio = it.numPassaggio
                    }
                )
            }
            newList
        }
    }

}
